### DISCUSSION TITLE
Error while installing Alive2

### DISCUSSION THREAD

**Discussion by kyledutyy** [AliveToolkit/alive2#699] [Q&A] [answered]
Hi, I was trying alive2, I followed the steps and installed all the required pakages. I am facing a error while building alive2, here is the error log.

`[1/45] Building CXX object CMakeFiles/tools.dir/tools/transform.cpp.o
FAILED: CMakeFiles/tools.dir/tools/transform.cpp.o 
/usr/bin/c++   -I../ -I. -Wall -Werror -march=native -fPIC  -O3 -O3    -std=c++2a -MD -MT CMakeFiles/tools.dir/tools/transform.cpp.o -MF CMakeFiles/tools.dir/tools/transform.cpp.o.d -o CMakeFiles/tools.dir/tools/transform.cpp.o -c ../tools/transform.cpp
In file included from ../smt/exprs.h:6,
                 from ../ir/attrs.h:6,
                 from ../ir/function.h:6,
                 from ../tools/transform.h:6,
                 from ../tools/transform.cpp:4:
../smt/expr.h:6:10: fatal error: compare: No such file or directory
    6 | #include <compare>
      |          ^~~~~~~~~
compilation terminated.

[2/45] Building CXX object CMakeFiles/smt.dir/smt/expr.cpp.o
FAILED: CMakeFiles/smt.dir/smt/expr.cpp.o 
/usr/bin/c++   -I../ -I. -Wall -Werror -march=native -fPIC  -O3 -O3    -std=c++2a -MD -MT CMakeFiles/smt.dir/smt/expr.cpp.o -MF CMakeFiles/smt.dir/smt/expr.cpp.o.d -o CMakeFiles/smt.dir/smt/expr.cpp.o -c ../smt/expr.cpp
In file included from ../smt/expr.cpp:4:
../smt/expr.h:6:10: fatal error: compare: No such file or directory
    6 | #include <compare>
      |          ^~~~~~~~~
compilation terminated.

[3/45] Building CXX object CMakeFiles/smt.dir/smt/exprs.cpp.o
FAILED: CMakeFiles/smt.dir/smt/exprs.cpp.o 
/usr/bin/c++   -I../ -I. -Wall -Werror -march=native -fPIC  -O3 -O3    -std=c++2a -MD -MT CMakeFiles/smt.dir/smt/exprs.cpp.o -MF CMakeFiles/smt.dir/smt/exprs.cpp.o.d -o CMakeFiles/smt.dir/smt/exprs.cpp.o -c ../smt/exprs.cpp
In file included from ../smt/exprs.h:6,
                 from ../smt/exprs.cpp:4:
../smt/expr.h:6:10: fatal error: compare: No such file or directory
    6 | #include <compare>
      |          ^~~~~~~~~
compilation terminated.

[4/45] Building CXX object CMakeFiles/smt.dir/smt/smt.cpp.o
FAILED: CMakeFiles/smt.dir/smt/smt.cpp.o 
/usr/bin/c++   -I../ -I. -Wall -Werror -march=native -fPIC  -O3 -O3    -std=c++2a -MD -MT CMakeFiles/smt.dir/smt/smt.cpp.o -MF CMakeFiles/smt.dir/smt/smt.cpp.o.d -o CMakeFiles/smt.dir/smt/smt.cpp.o -c ../smt/smt.cpp
In file included from ../smt/solver.h:6,
                 from ../smt/smt.cpp:6:
../smt/expr.h:6:10: fatal error: compare: No such file or directory
    6 | #include <compare>
      |          ^~~~~~~~~
compilation terminated.

[6/45] Building CXX object CMakeFil...ver.dir/tools/alive-jobserver.cpp.o
ninja: build stopped: subcommand failed.
`

I need some help regarding this.
Thanks.

**Comment by nunoplopes** [comment, marked as answer]
Your compiler is too old 😅  (I guess we should check this in cmake, but..)
You need either gcc 10 or clang 11 (or newer). Major distros already have packages for those.
